system/jlib/jmisc.cpp

```diff
@@ -696,21 +696,31 @@ class AbortHandlerInfo : public CInterface
 public:
     ThreadId installer;
     AbortHandler handler;
+    AbortHandlerBase bhandler;
     IAbortHandler *ihandler;
     AbortHandlerInfo(AbortHandler _handler)
     {
         handler = _handler;
+        bhandler = NULL;
+        ihandler = NULL;
+        installer = GetCurrentThreadId();
+    }
+    AbortHandlerInfo(AbortHandlerBase _handler)
+    {
+        handler = NULL;
+        bhandler = _handler;
         ihandler = NULL;
         installer = GetCurrentThreadId();
     }
     AbortHandlerInfo(IAbortHandler *_ihandler)
     {
         handler = NULL;
+        bhandler = NULL;
         ihandler = _ihandler;
         installer = GetCurrentThreadId();
     }
 
-    bool handle()
+    bool handle(ahType aht_val)
```

Comment:
Minor style: We generally use camelCase instead of underscores in identifiers.  A name of "type" would be easier to understand than aht_val.